pandas/core/strings/accessor.py

```diff
@@ -689,12 +690,15 @@ def cat(
         out: Index | Series
         if isinstance(self._orig, ABCIndex):
             # add dtype for case that result is all-NA
+            dtype = None
+            if isna(result).all():
```

Comment:
Does the comment in `test_str_cat_all_na` refer to the the object dtype here (othewise good to add a TODO comment here)

Reply:
Yes that's the place where I found this being an issue